Api/Resource.php

```diff
@@ -50,6 +50,34 @@ class Resource implements ResourceInterface
      */
     private $validationGroups;
     /**
+     * @var bool
+     */
+    private $paginationEnabledByDefault = false;
+    /**
+     * @var bool
+     */
+    private $clientAllowedToEnablePagination = false;
+    /**
+     * @var float
+     */
+    private $itemsPerPageByDefault = 30.;
```

Comment:
I'd say that 9,223,372,036,854,775,807 items per page starts to be a too much :sweat_smile: